File: src/WebUI/ClientApp/src/app/shared/models/paginated-list.ts
export class PaginatedList<T>{
    items: Array<T> = [];
    pageNumber: number = 0;
    totalPages: number = 10;
    totalCount: number = 0;
    pageSize: number = 10;

    constructor(items: Array<T>, count: number, totalPages: number, totalCount: number, pageSize: number) {
        this.items = items;
        this.totalCount = count;
        this.totalPages = totalPages;
        this.totalCount = totalCount;
        this.pageSize = pageSize;
    }
}

File: src/Application/Common/Models/PaginatedList.cs
using Microsoft.EntityFrameworkCore;

namespace WeCare.Application.Common.Models;
public class PaginatedList<T>
{
    public List<T> Items { get; }
    public int PageNumber { get; }
    public int TotalPages { get; }
    public int TotalCount { get; }
    public int PageSizee { get; }

    public PaginatedList(List<T> items, int count, int pageNumber, int pageSize)
    {
        PageNumber = pageNumber;
        TotalPages = (int)Math.Ceiling(count / (double)pageSize);
        TotalCount = count;
        Items = items;
        PageSizee = pageSize;
    }

    public PaginatedList(List<T> items, int totalPages, int pageNumber, int pageSize,int count)
    {
        PageNumber = pageNumber;
        TotalPages = totalPages;
        TotalCount = count;
        Items = items;
        PageSizee = pageSize;
    }
    public bool HasPreviousPage => PageNumber > 1;

    public bool HasNextPage => PageNumber < TotalPages;

    public static async Task<PaginatedList<T>> CreateAsync(IQueryable<T> source, int pageNumber, int pageSize)
    {
        var count = await source.CountAsync();
        var items = await source.Skip((pageNumber - 1) * pageSize).Take(pageSize).ToListAsync();

        return new PaginatedList<T>(items, count, pageNumber, pageSize);
    }
}
